Unhandled exception in vault_identity_group_member_entity_ids on plan

### Terraform Version
Terraform v1.1.6

### Affected Resource(s)
Please list the resources as a list, for example:
- vault_identity_entity_alias
- vault_identity_group_member_entity_ids

Plugin becomes unresponsive ("Plugin did not respond") after the first "vault_identity_entity_alias".
On repeated executions of `terraform plan`, the crash occurs at different resources (of this type).
99%-same config on staging server (another vault instance) works fine.

### Terraform Configuration Files
https://github.com/hashicorp/terraform-provider-vault/files/8102020/tf-config-20220219-gpg.zip

### Debug Output
https://gist.github.com/dietmarb01/3d243bb5183460ff870dbcf3fc055e9f

### Panic Output
panic: interface conversion: interface {} is nil, not []interface {}

```text
goroutine 303 [running]:
github.com/hashicorp/terraform-provider-vault/vault.identityGroupMemberEntityIdsRead(0xc000564c80, {0x10d8520, 0xc0009db180})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vault/vault/resource_identity_group_member_entity_ids.go:130 +0x75d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x10e5edc, {0x13586c8, 0xc0005e79c0}, 0x24, {0x10d8520, 0xc0009db180})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:346 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000177960, {0x13586c8, 0xc0005e79c0}, 0xc000744340, {0x10d8520, 0xc0009db180})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:635 +0x35b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0004bad80, {0x13586c8, 0xc0005e79c0}, 0xc0005e7a80)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/grpc_provider.go:576 +0x534
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc000565300, {0x1358770, 0xc0004e4510}, 0xc000077380)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:553 +0x3b0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x10a1ec0, 0xc000565300}, {0x1358770, 0xc0004e4510}, 0xc000077320, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000158a80, {0x136ea80, 0xc0000dbe00}, 0xc0004047e0, 0xc0004e4300, 0x1b35270, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:1279 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc000158a80, {0x136ea80, 0xc0000dbe00}, 0xc0004047e0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:1608 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:923 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:921 +0x294
```

### Expected Behavior
run to completion

### Actual Behavior
crash

### Steps to Reproduce

1. `terraform plan`

### Important Factoids

### References

I am seeing the same thing, but with different objects, and only during applies, not plans.  I even tried downgrading to 3.2.1 and still get the panic.

Only one object in the config seems to be causing it.  If I remove this object, i do not get the panic:

```hcl
# Configure snapshots of the raft integrated storage
resource "vault_raft_snapshot_agent_config" "default" {
  name             = "s3_daily"
  interval_seconds = 86400
  retain           = var.snapshot_retention_count
  path_prefix      = var.snapshot_bucket_path_prefix
  storage_type     = "aws-s3"

  aws_s3_bucket     = data.aws_s3_bucket.snapshots.bucket
  aws_s3_region     = data.aws_region.current.name
  aws_s3_enable_kms = true
  aws_s3_kms_key    = var.snapshot_kms_key_id
}

data "aws_s3_bucket" "snapshots" {
  bucket = var.snapshot_bucket
}

data "aws_region" "current" {}
```

Similar crash output as above:

```
module.vault_config.vault_raft_snapshot_agent_config.default: Creating...
╷
│ Error: Plugin did not respond
│
│   with module.vault_config.vault_raft_snapshot_agent_config.default,
│   on ../../../../../../../providers/vault/modules/vault-base-config/snapshots.tf line 2, in resource "vault_raft_snapshot_agent_config" "default":
│    2: resource "vault_raft_snapshot_agent_config" "default" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-vault_v3.3.0_x4 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1a8dc5a]

goroutine 85 [running]:
github.com/hashicorp/terraform-provider-vault/vault.readSnapshotAgentConfigResource(0xc0005ac100, {0x1cd5e40, 0xc0005d8280})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vault/vault/resource_raft_snapshot_agent_config.go:321 +0x31a
github.com/hashicorp/terraform-provider-vault/vault.createOrUpdateSnapshotAgentConfigResource(0x0, {0x1cd5e40, 0xc0005d8280})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vault/vault/resource_raft_snapshot_agent_config.go:298 +0x287
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x1b93740, {0x1f55b08, 0xc000724000}, 0x2, {0x1cd5e40, 0xc0005d8280})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:329 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0007f1340, {0x1f55b08, 0xc000724000}, 0xc0004ef790, 0xc000785d00, {0x1cd5e40, 0xc0005d8280})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:467 +0x871
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00000c738, {0x1f55b08, 0xc000724000}, 0xc0000a0c30)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/grpc_provider.go:977 +0xd8a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0001d7000, {0x1f55bb0, 0xc0005cc2a0}, 0xc00033bb90)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:603 +0x30e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1c9fa00, 0xc0001d7000}, {0x1f55bb0, 0xc0005cc2a0}, 0xc00091d500, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0006ca8c0, {0x1f6be60, 0xc00018f080}, 0xc0001537a0, 0xc00081c390, 0x27373e0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:1279 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0006ca8c0, {0x1f6be60, 0xc00018f080}, 0xc0001537a0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:1608 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:923 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:921 +0x294

Error: The terraform-provider-vault_v3.3.0_x4 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

Here is the trace log: https://gist.github.com/nomeelnoj/a4d126cf42ac7a7321b9af583339c9f3

--- a/vault/resource_identity_group_member_entity_ids.go
+++ b/vault/resource_identity_group_member_entity_ids.go
@@ -116,8 +116,12 @@ func identityGroupMemberEntityIdsRead(d *schema.ResourceData, meta interface{}) 
 		return err
 	}
 
-	d.Set("group_id", id)
-	d.Set("group_name", resp.Data["name"])
+	if err := d.Set("group_id", id); err != nil {
+		return err
+	}
+	if err := d.Set("group_name", resp.Data["name"]); err != nil {
+		return err
+	}
 
 	if d.Get("exclusive").(bool) {
 		respdata := resp.Data["member_entity_ids"]
@@ -127,13 +131,14 @@ func identityGroupMemberEntityIdsRead(d *schema.ResourceData, meta interface{}) 
 	} else {
 		userMemberEntityIds := d.Get("member_entity_ids").(*schema.Set).List()
 		newMemberEntityIds := make([]string, 0)
-		apiMemberEntityIds := resp.Data["member_entity_ids"].([]interface{})
-
-		for _, memberEntityId := range userMemberEntityIds {
-			if found, _ := util.SliceHasElement(apiMemberEntityIds, memberEntityId); found {
-				newMemberEntityIds = append(newMemberEntityIds, memberEntityId.(string))
+		if ids, ok := resp.Data["member_entity_ids"].([]interface{}); ok && len(ids) > 0 {
+			for _, memberEntityId := range userMemberEntityIds {
+				if found, _ := util.SliceHasElement(ids, memberEntityId); found {
+					newMemberEntityIds = append(newMemberEntityIds, memberEntityId.(string))
+				}
 			}
 		}
+
 		if err = d.Set("member_entity_ids", newMemberEntityIds); err != nil {
 			return fmt.Errorf("error setting member entity ids for IdentityGroupMemberEntityIds %q: %s", id, err)
 		}

